tests/director/tests/bundle_api_test.go

```diff
@@ -464,9 +468,63 @@ func TestAddBundle(t *testing.T) {
 	require.NoError(t, err)
 
 	assertions.AssertBundle(t, &bndlInput, &output)
+
+	parsedTargetURL, err := url.Parse(bndlInput.APIDefinitions[0].TargetURL)
+	require.NoError(t, err)
+	expectedTargetUrl := fmt.Sprintf("%s://%s", parsedTargetURL.Scheme, parsedTargetURL.Host)
+
+	appWithBaseURL := fixtures.GetApplication(t, ctx, certSecuredGraphQLClient, tenantId, application.ID)
+	assert.NotNil(t, appWithBaseURL.BaseURL)
+	assert.Equal(t, expectedTargetUrl, str.PtrStrToStr(appWithBaseURL.BaseURL))
+
 	saveExample(t, bundleRequest.Query(), "query bundle")
 }
 
+func TestAddBundleForApplicationWithAlreadySetBaseURL(t *testing.T) {
+	ctx := context.Background()
+
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+
+	baseURL := "https://compass.kyma.local/api"
+	application, err := fixtures.RegisterApplicationWithBaseURL(t, ctx, certSecuredGraphQLClient, baseURL, tenantId)
+	defer fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenantId, &application)
+	require.NoError(t, err)
+	require.NotEmpty(t, application.ID)
+
+	bndlInput := fixtures.FixBundleCreateInputWithRelatedObjects(t, "bndl-app-1")
+	bndl, err := testctx.Tc.Graphqlizer.BundleCreateInputToGQL(bndlInput)
+	require.NoError(t, err)
+
+	addBndlRequest := fixtures.FixAddBundleRequest(application.ID, bndl)
+	output := graphql.BundleExt{}
+
+	// WHEN
+	t.Log("Create bundle")
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, addBndlRequest, &output)
+
+	// THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, output.ID)
+	assertions.AssertBundle(t, &bndlInput, &output)
+	defer fixtures.DeleteBundle(t, ctx, certSecuredGraphQLClient, tenantId, output.ID)
+
+	bundleRequest := fixtures.FixBundleRequest(application.ID, output.ID)
+	bndlFromAPI := graphql.ApplicationExt{}
+
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, bundleRequest, &bndlFromAPI)
+	require.NoError(t, err)
+
+	assertions.AssertBundle(t, &bndlInput, &output)
+
+	parsedTargetURL, err := url.Parse(bndlInput.APIDefinitions[0].TargetURL)
+	require.NoError(t, err)
+	expectedTargetUrl := fmt.Sprintf("%s://%s", parsedTargetURL.Scheme, parsedTargetURL.Host)
+
+	appWithBaseURL := fixtures.GetApplication(t, ctx, certSecuredGraphQLClient, tenantId, application.ID)
+	assert.NotNil(t, appWithBaseURL.BaseURL)
+	assert.NotEqual(t, expectedTargetUrl, str.PtrStrToStr(appWithBaseURL.BaseURL))
```

Comment:
Can you add one more assert after that which which asserts that the returned `appWithBaseURL.BaseURL` is equal to `baseURL := "https://compass.kyma.local/api"` defined above?

Reply:
done